src/DiagramGenerator/Generator.php

```diff
@@ -47,7 +48,7 @@ public function buildDiagram(Config $config, $rootCacheDir, $boardTextureUrl, $p
     {
         $errors = $this->validator->validate($config);
         if (count($errors) > 0) {
-            throw new \Exception($errors->__toString());
+            throw new InvalidConfigException($errors->__toString());
```

Comment:
This shouldn't affect V2 in any way, because v2 catches all exceptions [here](https://github.com/ChessCom/chess/blob/master/src/Chess/DiagramBundle/Controller/FrontendController.php#L77-L79). It doesn't really matter what exception we throw here. It's the same message.